datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+
+    com_manager = data.pop("Commercial ship manager (IMO / Country / Date)")
+    if com_manager != "":
+        com_manager_parts = com_manager.split(" / ")
+        if len(com_manager_parts) == 3:
+            com_name_imo, com_country, com_date = com_manager_parts
+            if len(com_name_imo.split(" (")) == 2:
+                com_name, com_imo = com_name_imo.split(" (")
+            else:
+                overrides = lookup_override(context, com_name_imo, "com_manager")
+                com_name = overrides.get("name")
+                com_imo = overrides.get("registrationCode")
+
+            com_manager = context.make("LegalEntity")
+            com_manager.id = context.make_id(com_name, com_country)
+            com_manager.add("name", com_name)
+            com_manager.add("registrationNumber", com_imo)
+            com_manager.add("country", com_country)
+            com_manager.add("topics", "poi")
+            context.emit(com_manager, target=True)
+
+            com_rep = context.make("Representation")
+            com_rep.id = context.make_id(
+                vessel.id, "commercially managed by", com_manager.id
+            )
+            com_rep.add("client", vessel.id)
+            com_rep.add("agent", com_manager.id)
+            com_rep.add("role", "Commercial ship manager")
+            h.apply_date(com_rep, "startDate", com_date)
+
+            context.emit(com_rep)
+
+    safety_manager = data.pop(
+        "Ship Safety Management Manager (IMO / Country / Date)", None
+    )
+    if safety_manager is not None:
+        safety_manager_parts = safety_manager.split(" / ")
+        if len(safety_manager_parts) == 3:
+            safety_name_imo, safety_country, safety_date = safety_manager_parts
+            if len(safety_name_imo.split(" (")) == 2:
+                safety_name, safety_imo = safety_name_imo.split(" (")
+            else:
+                overrides = lookup_override(context, safety_name_imo, "com_manager")
+                safety_name = overrides.get("name")
+                safety_imo = overrides.get("registrationCode")
+
+            safety_manager = context.make("LegalEntity")
+            safety_manager.id = context.make_id(safety_name, safety_country)
+            safety_manager.add("name", safety_name)
+            safety_manager.add("registrationNumber", safety_imo)
+            safety_manager.add("country", safety_country)
+            safety_manager.add("topics", "poi")
+            context.emit(safety_manager, target=True)
+
+            safety_rep = context.make("Representation")
+            safety_rep.id = context.make_id(
+                vessel.id, "safety managed by", safety_manager.id
+            )
+            safety_rep.add("client", vessel.id)
+            safety_rep.add("agent", safety_manager.id)
+            safety_rep.add("role", "Ship Safety Management Manager")
+            h.apply_date(safety_rep, "startDate", safety_date)
+
+            context.emit(safety_rep)
```

Comment:
this looks ripe for factoring out to a dedicated function crawl_ship_safety_manager(context, vessel, com_manager)